kedro/io/core.py

```diff
@@ -407,7 +406,10 @@ def parse_dataset_definition(
         )
         logging.getLogger(__name__).warning(message, VERSION_KEY)
         del config[VERSION_KEY]
-    if config.pop(VERSIONED_FLAG_KEY, False):  # data set is versioned
+
+    is_tracking_ds = "kedro.extras.datasets.tracking" in class_obj.__module__
+    # dataset is versioned or in case of the tracking datasets they must be versioned by default
+    if config.pop(VERSIONED_FLAG_KEY, False) or is_tracking_ds:
```

Comment:
Actually looking at it like this, I agree with @AntonyMilneQB 's intuition here. Another idea that came to my head was what if we had an attribute on the dataset classes that we want to be always versioned, and in here we check if that property exists and is `True`. That way it's not tied to a specific type of dataset (can be tracking, can be anything). It does feel a bit like a deja-vu with `Runner` and `is_async` though.

I know @limdauto wanted to have a look at this too so tagging him here.

Reply:
Yes I would also prefer if the logic is in the datasets and not `core`, but couldn't quickly come up with a better proposal.. 

@lorenabalan the attribute sounds like a good alternative, but I'm sensing the "deja-vu" isn't necessarily a positive one?

Reply:
It was just that it was a property we added because a specific class needed it and wanted to be able to tell, without `isinstance` & the like, which class we're dealing with. I think it's still a better approach than checking module path, so I'm happy to go with that as a solution.

Reply:
Let's not redo the whole versioning logic right now, but can be a thought for the future. 😂